### PR TITLE
C12324 time series aggregations

### PR DESCRIPTION
**Pull request by ivanmalagon**
Related to: https://github.com/CartoDB/cartodb/issues/12324

____________
cartodb PR: https://github.com/CartoDB/cartodb/pull/12347
cartodb.js PR: https://github.com/CartoDB/cartodb.js/pull/1685
deep-insights.js PR: https://github.com/CartoDB/deep-insights.js/pull/555

### CODE REVIEW

File: src/dataviews/histogram-dataview-model.js [hunk 1]
```diff
@@ -239,8 +274,56 @@ module.exports = DataviewModelBase.extend({
...
+      aggregation: model.get('aggregation'),
+      bins: model.get('bins'),
+      end: model.get('end'),
+      error: model.get('error'),
+      start: model.get('start')
+    }, { silent: true });
+
+    var resetFilter = false;
+
+    if (this.get('column_type') === 'date' && _.has(this.changed, 'aggregation')) {
```
Comment (matallo) [inline src/dataviews/histogram-dataview-model.js]: can't we use a `||` for the same variable assignment?

File: src/dataviews/histogram-dataview/histogram-data-model.js [hunk 2]
```diff
@@ -14,9 +17,12 @@ module.exports = Model.extend({
   },
 
   url: function () {
-    var params = [
-      'bins=' + this.get('bins')
-    ];
+    var params = [];
+    if (this.get('column_type') === 'date' && this.get('aggregation')) {
+      params.push('aggregation=' + this.get('aggregation'));
+    } else if (this.get('bins')) {
```
Comment (matallo) [inline src/dataviews/histogram-dataview/histogram-data-model.js]: some times we do the double check if has the attribute, and also if column type is number, doesn't this apply here?

File: src/dataviews/histogram-dataview/histogram-data-model.js [hunk 3]
```diff
@@ -34,7 +40,12 @@ module.exports = Model.extend({
 
   initialize: function () {
     this.sync = BackboneAbortSync.bind(this);
-    this.bind('change:url change:bins', function () {
+    this.on('change:url', function () {
```
Comment (matallo) [inline src/dataviews/histogram-dataview/histogram-data-model.js]: isn't `this.listenTo` applicable here?

File: src/dataviews/histogram-dataview/histogram-data-model.js [hunk 4]
```diff
@@ -41,8 +41,11 @@ module.exports = Model.extend({
 
   initialize: function () {
     this.sync = BackboneAbortSync.bind(this);
+    this._initBinds();
+  },
+
+  _initBinds: function () {
     this.on('change:url', function () {
-      console.log('change:url > fetch()');
```
Comment (xavijam) [inline src/dataviews/histogram-dataview/histogram-data-model.js]: 😬

Comment (ivanmalagon) [inline src/dataviews/histogram-dataview/histogram-data-model.js]: Removed.

It was spotted before seeing your comment.

Comment (xavijam) [inline src/dataviews/histogram-dataview/histogram-data-model.js]: Que yaaaaaa

**Pull request merged**